aten/src/ATen/native/quantized/cpu/qconv.cpp

```diff
@@ -598,6 +598,11 @@ class QConvInt8 final {
           reinterpret_cast<uint8_t*>(qnnp_w_data),
           reinterpret_cast<int32_t*>(bias.data_ptr<c10::qint32>()));
       pack_w = pack_data.w.get();
+#ifdef C10_MOBILE
+      // On mobile, we release the original weight by freeing the underlying storage.
+      // Calling unpack after this will throw an assertion.
+      pack_data.orig_weight.unsafeGetTensorImpl()->release_resources();
```

Comment:
What happens if you try to access tensor that has done `release_resources`?

Reply:
Also I am wondering if explicitly opting in or out of this behavior might be better compared to ifdef. Because now you are forced to this behavior. 
The scenario I am worried about is this:
Currently XNNPACK related optimizations that are wrapped inside optimizeForMobile JIT pass, work only when you do mobile build. This is because XNNPACK is not enabled internally on server only on mobile builds. Thus I have a binary that is supposed to be built for mobile and this binary takes in your model and run optimizeForMobile JIT pass on it. As part of that it needs to run op whose output is a torchbind class, ConvPackedContext or something like that. Now the prepack op is removed and we have ConvPackedContext. When we serialize the optimized model ConvPackContext is serialized by fetching underlying original tensor. Since we build with QNNPACK for server as well (?) this may not be a problem. And the problem I described is also temporary. We are gonna enable XNNPACK in server too, but I dont know what other situation might come up. So I am wondering if we should do via ifdef or explicitly opting in.

Reply:
I am not sure I fully understand the problem here. But this change shouldn't affect XNNPACK (unless you're planning on doing something similar for XNNPACK and want a general solution for both). 

> Thus I have a binary that is supposed to be built for mobile and this binary takes in your model and run optimizeForMobile JIT pass on it. 

You mean takes in a serialized quantized model? Do we run this pass for quantized models as well?

Reply:
> What happens if you try to access tensor that has done `release_resources`?

Looks like you can still access the tensor (eg. `tensor.numel()` works, but if you try to access the underlying storage it will throw.

Reply:
Please do not call `release_resources`. The documentation says explicitly what this method is used for, and it is not for your use case.

```
  /**
   * Release (decref) storage, and any other external allocations.  This
   * override is for `intrusive_ptr_target` and is used to implement weak
   * tensors.
   */
  virtual void release_resources() override;
```

Reply:
@ezyang, what would be the best way to free a tensor along with underlying memory?

Reply:
Set it to nullptr :)

Reply:
@dzhulgakov isn't that what `release_resources()` is doing under the hood? How would setting it to nullptr free memory?